libmachine/libmachine.go

```diff
@@ -169,6 +169,11 @@ func (api *Client) performCreate(h *host.Host) error {
 
 	log.Infof("Provisioning with %s...", provisioner.String())
 	if err := provisioner.Provision(*h.HostOptions.SwarmOptions, *h.HostOptions.AuthOptions, *h.HostOptions.EngineOptions); err != nil {
+		// Don't check if Docker was installed successfully on no-op provisioner
+		if _, ok := err.(provision.NoOpError); ok {
+			log.Warn("no-op provisioning does nothing, docker not installed")
+			return nil
+		}
```

Comment:
I feel like we can just use string() to tell us that this is no docker so that we return early, rather than call provision, return an error, then check the error type. This can be placed above provisioning log, with logs that explain provisioning will be skipped. If this is done noOpError can be removed. I think this is preferable because it is not really an error. Also, I do not think the log should be a warning since it is expected behavior.

Reply:
Yeah, you're right. Simple is better.